graphistry/feature_utils.py

```diff
@@ -81,6 +81,7 @@ def assert_imported_cucat():
         logger.warning(  # noqa
                 "cuml and/or cudf not found, trying running"  # noqa
                 "`pip install rapids`"  # noqa
+                "or `pip install --extra-index-url=https://pypi.nvidia.com cuml-cu11 cudf-cu11`"  #noqa
```

Comment:
looks good

however... why is this `assert_...` except never actually asserts, just warns? is this being used incorrectly, or being called incorrectly, or it's just incorrectly named?

Reply:
1. for cu_cat I have [removed requirement for cudf/cuml ](https://github.com/graphistry/cu-cat/blob/master/setup.py#L22)so user can install how they want
2. advantage being [cpu fallback of cu_cat can be tested](https://github.com/graphistry/pygraphistry/blob/707b404c366bfb38e4dee5c934ce36fa8f8a316f/.github/workflows/ci.yml#L161)(called `test-gpu-umap` sadly since no gpu runner)
3. so here I reordered so warning if no cuml > loads sklearn, and only if that cant load will it error

- can we abandon #486 ? I've merged that and #511 here already

- fresh colab runs great on many demo datasets (used for cu_cat pytests), apart from strange phenomena on heavy text columns where the `degree` distribution becomes skewed narrower. @tanmoyio loading via `cudf.from_pandas()` seems to partially remedy this despite `make_safe_gpu_dataframes` also applied in `cu_cat._gap_encoder`